### PR TITLE
List: organization owners

### PR DESCRIPTION
**Pull request by mlbright**
* ghe console script for all org owners
    * This was Patrick's reverse engineering efforts
    * We may have a ghe-console script from GitHub support
    (https://support.enterprise.github.com/hc/en-us/requests/51994)
    * We need to remove service accounts
* scaffolding for showing the above results

### CODE REVIEW

**Comment by toddocon** [comment]
Does this only run on the appliance? I've re-created your files in my environment (to test) and it's not working. I pull all of my stats via remoteRun so I think in org-owners.sh this line presumes running on the appliance, yes?

`github-env bin/runner -e production`

**Comment by toddocon** [comment]
The error was:

`...
    self.executeScript(self.scriptPath("org-owners.sh"))
AttributeError: 'ReportOrgOwners' object has no attribute 'scriptPath'
`

So, in ReportOrgOwners.py, I changed this

`self.executeScript(self.scriptPath("org-owners.sh"))`

to this

`self.executeScript(os.path.join("scripts",` "org-owners.sh"))`

Now the data file is built. My output, though, is a table 10,221 rows long. There is one owner per line.

**Comment by mlbright** [comment]
@toddocon I fixed the ghe-console script (thank you GHE support!)

**Comment by mlbright** [comment]
I'm not sure why you need to specify 'scripts' and I don't.

File: docs/org-owners.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+layout: default
+title: Organization Owners
+permalink: /org-owners
+---
+<h3>
+Organization owners
```
Comment (pluehne) [inline docs/org-owners.html]: @mlbright: Can you make this title case (“Organization Owners”)?

File: updater/update-stats.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 from reports.ReportRepositoryHistory import *
 from reports.ReportTokenlessAuth import *
 from reports.ReportUsers import *
+from reports.ReportOrgOwners import *
```
Comment (pluehne) [inline updater/update-stats.py]: @mlbright: Could you please sort these alphabetically?

File: updater/update-stats.py [hunk 3]
```diff
@@ -82,6 +83,7 @@ def main():
 	ReportRepositoryHistory(configuration, dataDirectory, metaStats).update()
 	ReportTokenlessAuth(configuration, dataDirectory, metaStats).update()
 	ReportUsers(configuration, dataDirectory, metaStats).update()
+	ReportOrgOwners(configuration, dataDirectory, metaStats).update()
```
Comment (pluehne) [inline updater/update-stats.py]: @mlbright: Could you please sort these alphabetically?

**Comment by pluehne** [comment]
@toddocon: It’s weird that

```python
self.executeScript(self.scriptPath("org-owners.sh"))
```

didn’t work for you, but

```python
self.executeScript(os.path.join("scripts", "org-owners.sh"))
```

did. In fact, `self.scriptPath` should eventually do the same thing.

Also, other reports (such as for Git versions) use the same call. Are they broken for you as well?

**Comment by pluehne** [comment]
For future versions of this report, I’d like the following improvements:

- a search box to quickly find the owners of a specific organization (which, I think, is the typical use case of this report)
- links to the user accounts of the owners, just like the organizations themselves are linked

File: docs/org-owners.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+permalink: /org-owners
+---
+<h3>
+Organization Owners
+</h3>
+
+<div class="row">
+	<div class="col-main">
+		<div class="chart-container">
+		    <table class="table" data-url="{{ site.dataURL }}/org-owners.tsv"></table>
```
Comment (pluehne) [inline docs/org-owners.html]: @mlbright: You accidentally mixed spaces and tabs here.

File: docs/org-owners.html [hunk 5]
```diff
@@ -8,14 +8,14 @@
...
-		</div>
-	</div>
-	<div class="col-aside">
-		<div class="info-box">
-			<p>Owners for each GitHub organization.</p>
-		</div>
-	</div>
+    <div class="col-main">
+        <div class="chart-container">
+            <table class="table" data-url="{{ site.dataURL }}/org-owners.tsv"></table>
```
Comment (pluehne) [inline docs/org-owners.html]: Sorry, I should have been more specific. We’re using tabs in the other files, and it would be best to keep that consistent.

**Comment by pluehne** [comment]
As I don’t have write access to @mlbright’s branch, I [branched this pull request](https://github.com/Autodesk/hubble/tree/patrick/org-owners).

7e9ce8dac3ff762b47181994d3711950d02fc0f2 adds some demo data for [autodesk.github.io](), and be7a159581fc6f833d9e63239651751adc396f32 is a work-in-progress commit, where I started to support multiple owner names on a single line in the table (with links to their GitHub profiles).

Note that the updater report’s output format needs to be changed in the same way (instead of having one line for each (organization, owner) pair, just output a single line for each organization with all the owners comma-separated). Also, the column heading should be *owners* and not *owner* (singular).

**Comment by larsxschneider** [comment]
Superseded by #38

**Pull request closed without merging**